src/redis-check-aof.c

```diff
@@ -397,15 +399,21 @@ int fileIsManifest(char *filepath) {
                 break;
             } else {
                 printf("Cannot read file: %s\n", filepath);
+                fclose(fp);
                 exit(1);
             }
         }
 
-        /* Skip comments lines */
+        /* We will skip comments lines.
+         * At present, the manifest format is fixed, see aofInfoFormat.
+         * We will be returned directly as long as it encounters other items. */
         if (buf[0] == '#') {
             continue;
         } else if (!memcmp(buf, "file", strlen("file"))) {
             is_manifest = 1;
+        } else {
+            is_manifest = 0;
```

Comment:
I think it's fine to break straight here (no need to reset is_manifest). Suppose we have a manifest file that only the first line is correct, i.e. it starts with file, but the second line is wrong (i.e. the second line doesn't start with file, like a human modification or disk failure, etc.), then we should still think of it as a manifest file and not an old-stlye like AOF. We'll still parse it in the same way as multi-part aof, except we'll report a minifest format error in the aofLoadManifestFromFile function, but that's true.

Reply:
right, that is good idea, i am changing it, thanks